### ISSUE TITLE
Several Git-related problems (transient mode broken, magit-popup not completely removed...)

### ISSUE THREAD

**Issue by allentiak** [syl20bnr/spacemacs#12056]
#### Description :octocat:
Since mid-February 2019 https://github.com/magit/magit/pull/3728 replaced `magit-popup` with `transient`.
Spacemacs does not seem to work with it...

Simlar problem: https://github.com/magit/magit/issues/3747#issuecomment-465771260
Could this bug be related with https://github.com/syl20bnr/spacemacs/issues/11998 and or https://github.com/syl20bnr/spacemacs/issues/10942, perhaps?

Git works OK, both via Commandline and other graphical frontends.

#### Reproduction guide :beetle:
Modify a file into a Git repo, and save it. Try to perform the corresponding commit.

*Observed behaviour:* :eyes: :broken_heart:
- Start Emacs
- Modify a file. Save it. (`:w`)

Option a) using magit-popup dispatch (`SPC g m`)
- I get "Wrong type argument: commandp, magit-dispatch-popup"
- (Can't perform commit)

Option b) using transient:
- Use transient mode (`SPC g .`)
- Stage the file (`w `)
- Commit the file with popup (`c`)
- Activate "insert mode" (`i`)
- Type commit text ("example text")
- Exit "insert mode" (`ESC`)
- With editor finish (`, ,`)
I get "GitError! There was a problem with the editor '/usr/bin/emacsclient --socket-name=/tmp/emacs1000/server'" (see https://github.com/syl20bnr/spacemacs/issues/10942)
- (Commit not performed)
- (Spacemacs DOES NOT performs the commit)

*Expected behaviour:* :heart: :smile:
- Start Emacs
- Modify a file. Save it.
- Stage the file
- Commit the file/hunk
- Edit commit message
- "Finish with editor"
- (Spacemacs performs the commit via magit)

#### System Info :computer:
- OS: gnu/linux
- Emacs: 26.1
- Spacemacs: 0.300.0
- Spacemacs branch: develop (rev. 4b195dd)*
*I applied https://github.com/syl20bnr/spacemacs/pull/11403 on top of develop. (It does not affect this issue. I have been using this keyboard configuration for over a year now, without any problems.)
- Graphic display: t
- Distribution: spacemacs
- Editing style: vim
- Completion: helm
- Layers:
```elisp
(helm auto-completion emacs-lisp git
      (markdown :variables markdown-live-preview-engine 'vmd)
      multiple-cursors treemacs org
      (shell :variables shell-default-height 30 shell-default-position 'bottom)
      spell-checking syntax-checking
      (version-control :variables version-control-diff-tool 'git-gutter+ version-control-global-margin t)
      (clojure :variables cljr-warn-on-eval nil clojure-enable-sayid t clojure-enable-clj-refactor t)
      (parinfer :variables ensure t progn
                (setq parinfer-extensions
                      '(defaults pretty-parens evil smart-tab smart-yank)))
      (keyboard-layout :variables kl-layout 'colemak-neio-inverted)
      latex ess html)
```
- System configuration features: XPM JPEG TIFF GIF PNG RSVG IMAGEMAGICK SOUND GPM DBUS GSETTINGS NOTIFY ACL LIBSELINUX GNUTLS LIBXML2 FREETYPE M17N_FLT LIBOTF XFT ZLIB TOOLKIT_SCROLL_BARS GTK3 X11 THREADS LIBSYSTEMD LCMS2

#### Backtrace :paw_prints:
```
<<BACKTRACE IF RELEVANT>>
```

**Comment by allentiak** [comment]
It seems to be solved in latest `develop`, after regenerating `.spacemacs`.

**Comment by amiryal** [comment]
> It seems to be solved in latest `develop`, after regenerating `.spacemacs`.

FWIW, for me it was not a case of regenerating `~/.spacemacs`, but rather of updating packages (clicked `[Update Packages]` in the Spacemacs home buffer.)

**Comment by allentiak** [comment]
Closes #12056

**Comment by korantu** [comment]
Had problem with magit commit, getting `There was a problem with the editor`

I had to disable fill-column-indicator for magit commit to work :

```diff
modified   layers/+source-control/git/packages.el
@@ -12,7 +12,7 @@
 (setq git-packages
       '(
         evil-magit
-        fill-column-indicator
+        ;; fill-column-indicator
         gitattributes-mode
         gitconfig-mode

```
